transport/gdrive-transport/src/main/java/org/apache/airavata/mft/transport/gdrive/GDriveReceiver.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.airavata.mft.resource.service.GDriveResourceGetRequest;
+import org.apache.airavata.mft.resource.service.ResourceServiceGrpc;
+import org.apache.airavata.mft.secret.client.SecretServiceClient;
+import org.apache.airavata.mft.secret.service.GDriveSecret;
+import org.apache.airavata.mft.secret.service.GDriveSecretGetRequest;
+import org.apache.airavata.mft.secret.service.SecretServiceGrpc;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+
+public class GDriveReceiver implements Connector {
+
+    private static final Logger logger = LoggerFactory.getLogger(GDriveReceiver.class);
+
+    private GDriveResource gdriveResource;
+    private Drive drive;
+
+    @Override
+    public void init(String resourceId, String credentialToken, String resourceServiceHost, int resourceServicePort, String secretServiceHost, int secretServicePort) throws Exception {
+        ResourceServiceGrpc.ResourceServiceBlockingStub resourceClient = ResourceServiceClient.buildClient(resourceServiceHost, resourceServicePort);
+        this.gdriveResource = resourceClient.getGDriveResource(GDriveResourceGetRequest.newBuilder().setResourceId(resourceId).build());
+
+        SecretServiceGrpc.SecretServiceBlockingStub secretClient = SecretServiceClient.buildClient(secretServiceHost, secretServicePort);
+        GDriveSecret gdriveSecret = secretClient.getGDriveSecret(GDriveSecretGetRequest.newBuilder().setSecretId(credentialToken).build());
+
+        HttpTransport transport = GoogleNetHttpTransport.newTrustedTransport();
+        JsonFactory jsonFactory = new JacksonFactory();
+        String jsonString = gdriveSecret.getCredentialsJson();
+        GoogleCredential credential = GoogleCredential.fromStream(new ByteArrayInputStream(jsonString.getBytes(StandardCharsets.UTF_8)), transport, jsonFactory);
+        if (credential.createScopedRequired()) {
+            Collection<String> scopes = DriveScopes.all();
+            credential = credential.createScoped(scopes);
+        }
+        drive = new Drive.Builder(transport, jsonFactory, credential).build();
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    @Override
+    public void startStream(ConnectorContext context) throws Exception {
+        logger.info("Starting GDrive Receiver stream for transfer {}", context.getTransferId());
+
+        String id = null;
+        FileList fileList = drive.files().list()
+                .setQ("name = '"+gdriveResource.getResourcePath()+"'")
+                .setFields("files(id,name,modifiedTime,md5Checksum,size)")
+                .execute();
+
+        for (File f : fileList.getFiles()) {
+                id = f.getId();
+        }
+
+        if (id == null) {
+            throw new IllegalStateException("GDrive Receiver was unable to retrieve the resource");
```

Comment:
Provide the resource id in the exception text.

Reply:
Hi Dimuthu,
Just to be clear,
1. Do you mean that the key in resources.json should be named something specific to google drive so that it is more meaningful (eg: instead of "resourcePath", maybe name it as "gdriveFileName").
2. In secrets.json instead of naming the key as "credentialsJson" change it to maybe "clientSecret"